graph_snyk_cve_sync.py

```diff
@@ -0,0 +1,358 @@
+                        if is_relation_applicable("=", ver, rule[0]['val']):
+                            affected.append(ver)
+                    elif '=' in key1:
+                        if is_relation_applicable("=", ver, rule[1]['val']):
+                            affected.append(ver)
+    return list(set(affected))
+
+
+def get_affected_versions_npm(rules, versions):
+    """Get affected versions for npm."""
+    affected = []
+    for ver in versions:
+        for rule in rules:
+            if Version(ver) in rule:
+                affected.append(ver)
+                break
+    return list(set(affected))
+
+
+def get_semver_versions(versions):
+    """Convert to semver version format."""
+    semver = []
+    for ver in versions:
+        semver.append(api.to_semver(ver))
+    return semver
+
+
+def retrieve_dict(object_key):
+    """Retrieve a dictionary stored as JSON from S3."""
+    return json.loads(retrieve_blob(object_key).decode())
+
+
+def retrieve_blob(object_key):
+    """Retrieve remote object content."""
+    return s3_resource.Object(CVE_BUCKET, object_key).get()['Body'].read()
+
+
+def read_data_from_s3(date):
+    """Read the snyk data from S3."""
+    try:
+        filename = "snyk-feed/" + date + ".json"
+        return retrieve_dict(filename)
+    except Exception as e:
+        logger.error(e)
+        return False
+
+
+def extract_data_from_feed(data):
+    """Fetch all the required info from the feed."""
+    for eco in data:
+        if eco == "java":
+            parse_maven_data(data[eco])
+        elif eco == "js":
+            parse_npm_data(data[eco])
+        elif eco == "python":
+            parse_pypi_data(data[eco])
+
+
+def parse_maven_data(maven_data):
+    """Parse data for maven."""
+    if len(maven_data) != 0:
+        CVE_DATA['maven'] = {}
+        for data in maven_data:
+            pkg = data['package']
+            try:
+                versions = get_versions_for_ep('maven', pkg)
+                if versions:
+                    if pkg not in CVE_DATA['maven']:
+                        CVE_DATA['maven'][pkg] = []
+                    data['ecosystem'] = "maven"
+                    vuln_versions = get_semver_versions(data['vulnerableVersions'])
+                    data['rules'] = get_version_rules(vuln_versions)
+                    data['affected'] = get_affected_versions(data['rules'], versions)
+                    del data['rules']
+                    data['description'] = re.sub("[\'\"]", "", data['description'])
+                    if len(data['cves']) == 0:
+                        data['pvtVuln'] = True
+                    else:
+                        data['pvtVuln'] = False
+                    CVE_DATA['maven'][pkg].append(data)
+            except ValueError:
+                logger.error("Encountered an error while trying to fetch versions for maven -> {}".
+                             format(pkg))
+    logger.info("----------------Maven Data-------------------")
```

Comment:
This is also Okay, but Just found more Classi way for this `"Maven Data".center(21, '-')` :)

Reply:
Nice. Wasnt aware of this